spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/PubsubAdmin.java

```diff
@@ -46,7 +47,8 @@
 	private SubscriptionAdminClient subscriptionAdminClient;
 
 	/** Default inspired in the subscription creation web UI. */
-	private static final int DEFAULT_ACK_DEADLINE = 10;
+	@Value("${spring.cloud.gcp.pubsub.subscriber.ackDeadline:10}")
```

Comment:
No, that's wrong. 
To have `@Value` resolved your must have specific configuration in the application context.
Also this looks like a mix of auto-config and regular components concerns.
This is fully fine to have this property resolution in the auto-config, but here it must be setter.